Source/WebKit/WPEPlatform/wpe/WPEView.cpp

```diff
@@ -172,9 +174,15 @@ static void wpeViewConstructed(GObject* object)
     G_OBJECT_CLASS(wpe_view_parent_class)->constructed(object);
 
     // FIXME: add API to set the default view size.
-    auto* priv = WPE_VIEW(object)->priv;
-    priv->width = 1024;
-    priv->height = 768;
+    auto* view = WPE_VIEW(object);
+    auto* priv = view->priv;
+    auto settings = wpe_display_get_settings(wpe_view_get_display(view));
+
+    GUniqueOutPtr<GError> error;
+    auto width = wpe_settings_get_uint32(settings, WPE_SETTING_TOPLEVEL_WIDTH, &error.outPtr());
+    g_assert(!error.get());
+    priv->width = width;
+    priv->height = wpe_settings_get_uint32(settings, WPE_SETTING_TOPLEVEL_HEIGHT, &error.outPtr());
```

Comment:
The getter only fails if the setting doesn't exist, and this is a builtin setting, so I think we can just ignore the error and pass nullptr.